### ISSUE TITLE
Ouput only Somatic option doesn't seem to work in the latest revision of var2vcf_paired.pl

### ISSUE THREAD

**Issue by OgnjenSB** [AstraZeneca-NGS/VarDict#48]
In the last revision (and perhaps some earlier ones) the option -M in var2vcf_paired.pl doesn't seem to have any effect on the output. I cross-tested the outputs from the new previous tool with an older version, and the old TSV output from TestSomatic.R with the new tool, and all runs output STATUS=Germline mutations as well.

Do you mind taking a look?

Thanks in advance!

P.S. If I can help in any way, please tell me!

**Comment by mjafin** [comment]
Hi @OgnjenSB do the STATUS=Germline calls have a filter field set at all?

**Comment by OgnjenSB** [comment]
Yes, they do.

**Comment by mjafin** [comment]
So they are soft filtered? 
I think the `-M` may be a misnomer and isn't supposed to hard filter anything out.

**Comment by OgnjenSB** [comment]
Correct, they are soft filtered. The script used to remove all but somatic mutations in the previous versions. Also, in the code there is evidence of revisions at that particular line which changed the behavior (I know for sure it worked correctly in 1.4.6). 

Although this can be manually or otherwise edited, I assume this is a bug.

**Comment by mjafin** [comment]
We took a decision to soft filter instead of hard filter here as it's fairly trivial to hard filter soft filtered variants (but not the other way around).

**Comment by OgnjenSB** [comment]
We might have misunderstood each other. These are lines from the VCF that has the option -M set:

1	1247589	.	C	G	95	SN1.5;P0.05	STATUS=Germline;SAMPLE=TUMOR;TYPE=SNV;SHIFT3=0;MSI=4.000;MSILEN=1;SSF=0.3;SOR=0;LSEQ=CCAAGGGAGTGGGGGGGGGG;RSEQ=GGGGCCGGGCGCCCACCTCT	GT:DP:VD:ALD:RD:AD:AF:BIAS:PMEAN:PSTD:QUAL:QSTD:SBF:ODDRATIO:MQ:SN:HIAF:ADJAF:NM	1/0:9:7:6,1:2,0:2,7:0.7778:0,2:13.4:1:24.7:1:1:0:60:1.333:0.8:0:1	1/1:7:7:6,1:0,0:0,7:1:0,2:7.1:1:34:1:1:0:57.1:6:1:0:0.7
1	3701663	.	T	C	170	PASS	STATUS=Germline;SAMPLE=TUMOR;TYPE=SNV;SHIFT3=0;MSI=5.000;MSILEN=1;SSF=0.04913;SOR=2.08948;LSEQ=TGCAGCACCTTGAGGTCCTG;RSEQ=GGGGGCCGGGCGCAGTACAG	GT:DP:VD:ALD:RD:AD:AF:BIAS:PMEAN:PSTD:QUAL:QSTD:SBF:ODDRATIO:MQ:SN:HIAF:ADJAF:NM	1/0:80:41:16,25:16,23:39,41:0.5125:2,2:20.6:1:31.9:1:1:1.08582:60:40:0.5195:0:1.4	0/1:39:13:11,2:12,14:26,13:0.3333:2,2:20.8:1:33.7:1:0.03714:6.12:60:12:0.3243:0:1.2

So when you asked if the filter is set AT ALL, I said yes, meaning that filtration at that field works normally. But no, apart from the STATUS=Germline in the INFO field, nothing is added to the line.

In short, the -M option doesn't do anything. It has no effect on the output whatsoever.

**Comment by mjafin** [comment]
Hi @OgnjenSB, the fact that one of the variants has the `P0.05` filter indicates to me that the filter here https://github.com/AstraZeneca-NGS/VarDict/blob/master/var2vcf_paired.pl#L177 is working for the `-M` option. If you run the same variant in tumor-only mode can you confirm that it doesn't get the `P0.05` filter?

The other variant is a bit funny. Wondering if @zhongwulai could comment further but it looks like it may be a complex variation site..? The genotypes for the tumour and normal seem inverted, that is VarDict seems to think that the variant is on different alleles for the tumour and normal, if I'm not completely mistaken. 

In bcbio we tend to additionally set a `REJECT` filter to any variant that has `STATUS=germline`.
